src/coprocessor/dag/expr/builtin_string.rs

```diff
@@ -944,6 +945,71 @@ impl ScalarFunc {
             None => Ok(Some(0)),
         }
     }
+
+    // See https://dev.mysql.com/doc/refman/5.6/en/string-functions.html#function_insert
+    #[inline]
+    pub fn insert_binary<'a, 'b: 'a>(
+        &'b self,
+        ctx: &mut EvalContext,
+        row: &'a [Datum],
+    ) -> Result<Option<Cow<'a, [u8]>>> {
+        let s: Cow<[u8]> = try_opt!(self.children[0].eval_string(ctx, row));
+        let pos = try_opt!(self.children[1].eval_int(ctx, row));
+        let mut len = try_opt!(self.children[2].eval_int(ctx, row));
+        let new_str: Cow<[u8]> = try_opt!(self.children[3].eval_string(ctx, row));
+
+        let s_len = s.len() as i64;
+        if pos < 1 || pos > s_len {
+            return Ok(Some(s));
+        }
+        if len < 0 || len > s_len - pos + 1 {
```

Comment:
How about using a variable for  `s_len - pos + 1`?

Reply:
updated.